docs/metadata.md

```diff
@@ -0,0 +1,582 @@
+generic purls pointing to the repository or bare url where the dependency came from.
+
+<details>
+<summary>Example - NPM dependency from a URL</summary>
+
+```json
+{
+  "name": "fecha",
+  "version": "https://github.com/taylorhakes/fecha/archive/91680e4db1415fea33eac878cfd889c80a7b55c7.tar.gz",
+  "type": "npm"
+}
+```
+```
+pkg:generic/fecha?download_url=https%3A%2F%2Fgithub.com%2Ftaylorhakes%2Ffecha%2Farchive%2F91680e4db1415fea33eac878cfd889c80a7b55c7.tar.gz
+```
+</details>
+
+<details>
+<summary>Example - pip dependency from a git repo</summary>
+
+```json
+{
+  "name": "dockerfile-parse",
+  "version": "git+https://github.com/containerbuildsystem/dockerfile-parse@ed38dc2eadf6890b524340f45a7d172d02196e61",
+  "type": "pip"
+}
+```
+```
+pkg:github/containerbuildsystem/dockerfile-parse@ed38dc2eadf6890b524340f45a7d172d02196e61
+```
+</details>
+
+#### Local dependencies
+
+Often, package managers also support using local files as dependencies, see
+[Nested Dependencies][feature-support] in the README. For local dependencies, Cachito
+constructs the purl by taking the purl of the package that uses that dependency and
+appending the relative path from the package to the dependency to the package purl.
+
+<details>
+<summary>Example</summary>
+
+The package
+
+```json
+{
+  "name": "k8s.io/kubernetes",
+  "version": "v1.22.0",
+  "type": "gomod"
+}
+```
+```
+pkg:golang/k8s.io%2Fkubernetes@v1.22.0
+```
+
+The local dependency
+
+```json
+{
+  "name": "k8s.io/kubectl",
+  "version": "./staging/src/k8s.io/kubectl",
+  "type": "gomod"
+}
+```
+```
+pkg:golang/k8s.io%2Fkubernetes@v1.22.0#staging/src/k8s.io/kubectl
+```
+</details>
+
+In the Golang world with its modules and packages this is, as usual, more complicated.
+The example above pretends that `k8s.io/kubernetes` is a package and uses
+`k8s.io/client-go` as a dependency. That is not true - `k8s.io/kubernetes` is a *module*
+and contains many *packages*, but is not a package itself (you can tell by the absence
+of `*.go` files in the root of the [repo](https://github.com/kubernetes/kubernetes)).
+A more realistic example would be to use `k8s.io.kubernetes/cmd/kubeadm` as the package.
+The purls would look as follows:
+
+```
+# k8s.io/kubernetes/cmd/kubeadm
+pkg:golang/k8s.io%2Fkubernetes%2Fcmd%2Fkubeadm@v1.22.0
+# k8s.io/client-go
+pkg:golang/k8s.io%2Fkubernetes@v1.22.0#staging/src/k8s.io/kubectl
+```
+
+But wait, why is the purl for `k8s.io/client-go` relative to `k8s.io/kubernetes`, not
+to `k8s.io.kubernetes/cmd/kubeadm`? Dependency management is fully up to the module.
+All the dependencies, including local ones, are defined in the go.mod file in the
+module. The paths are likewise relative to the module, not to the package that ends up
+using the dependencies. That is why purls for local dependencies are relative to modules
+and not packages.
+
+## Request JSON vs. Content Manifest
+
+Both formats attempt to represent similar information about a Cachito request. Both do
```

Comment:
Is it naive to think that if we simply add `purl` attributes to each package/dependency in the `request-json` manifest format, then we have the best of both worlds?

Reply:
That is a surprisingly simple solution. It might actually work :open_mouth: 

I was under the impression that Content Manifest was the intended winner of the unification efforts, but if that is not the case, then this might just be all we need.